R/readKane.R

```diff
@@ -0,0 +1,63 @@
+#'
+#' @return a list with meta data and raw table reads. This will be 49 KB in size.
+#'
+#' @export
+
+dataDir <- '~/Desktop/Research/Kane' #edit to reflect personal directory
+
+
+#data read in for citation [2] not included
+  #url confirmation is needed, see ##TODO from 114_Kane.Rmd
+#KaneBiophysicalData is missing header information that could not be found online, contacting of researchers is necessary
+
+
+readKane2004 <- function(dataDir, download = TRUE, verbose = FALSE){
+ 
+  
+  url_table <- data.frame(File_Base = c('KaneSoilData.txt', 'KaneBiophysicalData.txt', 'KaneMetaData.xml'),
+                          Download_URL = c('http://www.lter.uaf.edu/php/download_data.php?f=/data_files/ascii/files/190_2031_all_soil_profile_depths_carbon_BD.txt', 'http://www.lter.uaf.edu/php/download_data.php?f=/data_files/ascii/files/190_1608_sitesummarydata.txt', 'http://www.lter.uaf.edu/eml/BNZeml132.xml'))
+  
+  
+  #for loop to read in data from url_table if it does not exist in local repository
+  for(ii in 1:nrow(url_table)){
+    DataFile <- file.path(dataDir, url_table$File_Base[ii])
+    if(!(file.exists(DataFile))){
+      download.file(url_table$Download_URL[ii], destfile= DataFile, quiet=FALSE)
+    }
+  }
+  
+  # #this code was not used as it requires additional package dependency
+  # plyr::d_ply(
+  #   url_table,
+  #   c('Download_URL','File_Base'),
+  #   function(xx){
+  #     DataFile <- file.path(dataDir, xx$File_Base)
+  #     if(!(file.exists(DataFile))){
+  #     download.file(xx$Download_URL, destfile= DataFile, quiet=FALSE)
+  #     }
+  #     }
+  #   )
+  
+  #reading in data for R
+  readKanePitData <- read_csv(file.path(dataDir, 'KaneSoilData.txt'))
+  readKaneBiophysicalData <- read_csv(file.path(dataDir, 'KaneBiophysicalData.txt'), col_names = c("Site Description", "Site ID", "X3", "X4", "X5", "X6", "X7", "X8", "X9", "X10", "X11", "X12", "X13", "X14")) #note that most headers are missing
+
+  #ans and its return
+  ans <- list(downloadFiles = c(file.path(dataDir,'KaneSoilData', 'KaneBiophysicalData', 'KaneMetaData')),
+              licenseShort = c('NA', 'NA', 'NA'),
+              licenseFull = c('NA', 'NA', 'NA'),
+              citation = c("Kane, Evan S.; Ping, Chien-Lu L. 2004. Soil carbon stabilization along productivity gradients in interior Alaska: Summer 2003, Bonanza Creek LTER - University of Alaska Fairbanks. BNZ:132, http://www.lter.uaf.edu/data/data-detail/id/132.", "Kane, Evan S.; Ping, Chien-Lu L. 2004. Soil carbon stabilization along productivity gradients in interior Alaska: Summer 2003, Bonanza Creek LTER - University of Alaska Fairbanks. BNZ:132, http://www.lter.uaf.edu/data/data-detail/id/132.", "Kane, Evan S.; Ping, Chien-Lu L. 2004. Soil carbon stabilization along productivity gradients in interior Alaska: Summer 2003, Bonanza Creek LTER - University of Alaska Fairbanks. BNZ:132, http://www.lter.uaf.edu/data/data-detail/id/132."),
```

Comment:
If all the citations are exactly the same then you should only include one citation and license. Treat this as a data package.